src/clr-angular/forms/radio/radio-container.ts

```diff
@@ -4,23 +4,86 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Component } from '@angular/core';
+import { Component, ContentChild, Input, OnDestroy, Optional } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { isBoolean } from 'util';
 
-import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
-import { ControlIdService } from '../common/providers/control-id.service';
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { ClrLabel } from '../common/label';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { NgControlService } from '../common/providers/ng-control.service';
 
 @Component({
   selector: 'clr-radio-container',
   template: `
-        <!-- We want the radio input to be before the label, always -->
-        <ng-content select="[clrRadio]"></ng-content>
-        <ng-content></ng-content>
-        <label *ngIf="_dynamic"></label>
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [ngClass]="controlClass()">
+      <div class="clr-radio-wrapper" [class.clr-radio-inline]="clrInline">
+        <ng-content select="clr-radio-wrapper"></ng-content>
+      </div>
+      <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+      <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+      <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+    </div>
     `,
-  host: { '[class.radio]': 'true' },
-  providers: [ControlIdService],
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [NgControlService, ControlClassService, IfErrorService],
 })
-export class ClrRadioContainer implements DynamicWrapper {
-  // Indicates whether the container is dynamically created by the radio button itself
-  _dynamic = false;
+export class ClrRadioContainer implements OnDestroy {
+  private subscriptions: Subscription[] = [];
+  invalid = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
+  private inline = false;
+
+  /*
+   * Here we want to support the following cases
+   * clrInline - true by presence
+   * clrInline="true|false" - unless it is explicitly false, strings are considered true
+   * [clrInline]="true|false" - expect a boolean
+   */
+  @Input()
+  set clrInline(value: boolean | string) {
+    if (!isBoolean(value)) {
+      this.inline = value === 'false' ? false : true;
+    } else {
+      this.inline = !!value;
+    }
+  }
+  get clrInline() {
```

Comment:
Nit: Do we need `clr` prefix here for this internal property? I think It's inconsistent with what we do currently. If we use setter/getter, we make the private property have an underscore and expose it with getter without one instead of prefixing the getter with `clr`.

Reply:
I did this because the `clrInline` is the name of the Input, and the `inline` property is private so I needed something. I thought it was a little different from standard setter/getter examples because clrInline is now a property of the component that gets exposed publicly, and I didn't want to confuse with a second property of the same name.